src/communication/scrollsguide.py

```diff
@@ -87,37 +106,42 @@ def types(self) -> str:
     def kind(self) -> str:
         return self._kind.capitalize()
 
-
-class ScrollNotFound(Exception):
-    pass
-
-
-class MultipleScrollsFound(Exception):
-    def __init__(self, scrolls, search_term):
-        self.scrolls = [scroll['name'] for scroll in scrolls]
-        self.search_term = search_term
-
-    def __str__(self):
-        if len(self.scrolls) > 5:
-            return f"Too many scrolls start with {self.search_term}"
-        return f'Multiple scrolls starting with {self.search_term}: {", ".join(self.scrolls)}'
-
-
-async def get_scroll(name: str) -> Scroll:
-    """Gets information about a scroll"""
-    async with aiohttp.ClientSession() as s:
-        # I'm sorry for fetching all of them, but the only limiting param is Id and I don't have the dict for that yet
-        async with s.get('http://a.scrollsguide.com/scrolls') as resp:
-            text = await resp.text()
-            data = json.loads(text)
-            backup_scrolls = list()
-            for scroll in data.get('data', []):
-                if scroll['name'].lower() == name.lower():
-                    return Scroll(scroll)
-                elif scroll['name'].lower().startswith(name.lower()):
-                    backup_scrolls.append(scroll)
-            if len(backup_scrolls) == 1:
-                return Scroll(backup_scrolls[0])
-            elif backup_scrolls:
-                raise MultipleScrollsFound(backup_scrolls, name)
-            raise ScrollNotFound
+    @classmethod
+    async def load_scrolls(cls: Type['Scroll']) -> None:
+        """Gets information about scrolls"""
+        async with aiohttp.ClientSession() as s:
+            async with s.get('http://a.scrollsguide.com/scrolls') as resp:
+                text = await resp.text()
+                data = json.loads(text)
+                cls._scrolls_db = {}
+                cls._scrolls_names = {}
+                for scroll_data in data.get('data', []):
+                    scroll = Scroll(scroll_data)
+                    cls._scrolls_db[scroll._id] = scroll
+                    cls._scrolls_names[scroll.name.lower()] = scroll._id
+
+    @classmethod
+    async def get(cls: Type['Scroll'], query: str, threshold: float = 0.7) -> 'Scroll':
```

Comment:
Let's call it `get_by_name`, `from_name`, `by_name` or something like that :)

Reply:
Ok